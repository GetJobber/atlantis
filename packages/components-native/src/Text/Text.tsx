import React from "react";
import {
  BaseWeight,
  LineHeight,
  TextAccessibilityRole,
  TextAlign,
  TextSize,
  TextVariation,
  TruncateLength,
  Typography,
  TypographyProps,
} from "../Typography";
import { tokens } from "../utils/design";
import { TypographyUnsafeStyle } from "../Typography/Typography";

export interface TextProps
  extends Pick<TypographyProps<"base">, "maxFontScaleSize" | "selectable"> {
  /**
   * Visual hierarchy of the text
   */
  readonly level?: TextLevel;

  /**
   * Color variation of text
   */
  readonly variation?: TextVariation;

  /**
   * Change the appearance of the text
   */
  readonly emphasis?: "strong";

  /**
   * Allow text to be resized based on user's device display scale
   */
  readonly allowFontScaling?: boolean;

  /**
   * Determines whether text should be scaled down to fit based on maxLines prop
   */
  readonly adjustsFontSizeToFit?: boolean;

  /**
   * The maximum amount of lines the text can occupy before being truncated with "...".
   * Uses predefined string values that correspond to a doubling scale for the amount of lines.
   */
  readonly maxLines?: TruncateLength;

  /**
   * Alignment of text
   */
  readonly align?: TextAlign;

  /**
   * Text to display
   */
  readonly children?: string;

  /**
   * Reverse theme for better display on dark background
   */
  readonly reverseTheme?: boolean;

  /**
   * Have text styled with strike through
   */
  readonly strikeThrough?: boolean;

  /**
   * Use italic font style
   */
  readonly italic?: boolean;

  /**
   * Underline style to use for the text. The non-solid style is only supported
   * on iOS, as per React Native's Text component's limitations.
   * https://reactnative.dev/docs/text-style-props#textdecorationstyle-ios
   */
  readonly underline?: "solid" | "dotted";

  /**
   * This will make the text inaccessible to the screen reader.
   * This should be avoided unless there is a good reason.
   * For example this is used in InputText to make it so the label isn't
   * selectable because it is already read from the accessibilityLabel
   * of the TextInput
   */
  readonly hideFromScreenReader?: boolean;

  /**
   * **Use at your own risk:** Custom style for specific elements. This should only be used as a
   * **last resort**. Using this may result in unexpected side effects.
   * More information [here](https://atlantis.getjobber.com/storybook/?path=/docs/guides-customizing-components--docs#unsafe_-props).
   */
  readonly UNSAFE_style?: TypographyUnsafeStyle;
}

export type TextLevel = "text" | "textSupporting";

interface LevelStyle {
  readonly size: TextSize;
  readonly lineHeight: LineHeight;
}

const levelStyles: Record<TextLevel, LevelStyle> = {
  text: {
    size: "default",
    lineHeight: "base",
  },
  textSupporting: {
    size: "small",
    lineHeight: "tight",
  },
};

const MAX_TEXT_FONT_SIZE_SCALE = 50;
export const TEXT_MAX_SCALED_FONT_SIZES: Record<TextLevel, number> = {
  text: MAX_TEXT_FONT_SIZE_SCALE,
  textSupporting: tokens["typography--fontSize-base"],
};

export function Text({
  level = "text",
  variation = "base",
  emphasis,
  allowFontScaling = true,
  adjustsFontSizeToFit = false,
  maxLines = "unlimited",
  align,
  children,
  reverseTheme = false,
  strikeThrough = false,
  italic = false,
  hideFromScreenReader = false,
  maxFontScaleSize,
  UNSAFE_style,
  underline,
  selectable,
}: TextProps): JSX.Element {
  const accessibilityRole: TextAccessibilityRole = "text";

  return (
    <Typography
      color={variation}
      UNSAFE_style={UNSAFE_style}
      fontFamily="base"
      fontStyle={italic ? "italic" : "regular"}
      fontWeight={getFontWeight({ level, emphasis })}
      maxFontScaleSize={maxFontScaleSize || TEXT_MAX_SCALED_FONT_SIZES[level]}
      selectable={selectable}
      underline={underline}
      {...{
        ...levelStyles[level],
        allowFontScaling,
        align,
        adjustsFontSizeToFit,
        accessibilityRole,
        reverseTheme,
        maxLines,
        strikeThrough,
        italic,
        hideFromScreenReader,
      }}
    >
      {children}
    </Typography>
  );
}

function getFontWeight({
  level,
  emphasis,
}: Pick<TextProps, "level" | "emphasis">): BaseWeight {
  if (emphasis === "strong") return "semiBold";
  if (level === "textSupporting") return "medium";

  return "regular";
}
