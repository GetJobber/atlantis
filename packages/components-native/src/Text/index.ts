export { Text, TextLevel } from "./Text";
