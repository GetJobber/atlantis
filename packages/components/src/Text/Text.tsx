import React, { PropsWithChildren } from "react";
import { Typography, TypographyOptions } from "../Typography";

interface TextProps {
  readonly variation?:
    | "default"
    | "subdued"
    | "success"
    | "error"
    | "warn"
    | "info";

  readonly size?: "small" | "base" | "large";
}

type TextColor = Extract<TypographyOptions, "textColor">;

export function Text({
  variation = "default",
  size = "base",
  children,
}: PropsWithChildren<TextProps>) {
  const textColors = {
    default: "greyBlueDark",
    subdued: "greyBlue",
    success: "green",
    error: "red",
    warn: "yellowDark",
    info: "lightBlue",
  };

  return (
    <Typography textColor={textColors[variation] as TextColor} size={size}>
      {children}
    </Typography>
  );
}
