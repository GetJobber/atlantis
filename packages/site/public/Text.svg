<svg width="224" height="117" viewBox="0 0 224 117" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M82 27C82 22.5817 85.5817 19 90 19H135C139.418 19 143 22.5817 143 27V91C143 95.4183 139.418 99 135 99H90C85.5817 99 82 95.4183 82 91V27Z" fill="white"/>
<path d="M97.0071 30.6136V29.3636H105.734V30.6136H102.075V41H100.666V30.6136H97.0071ZM110.121 41.1818C109.28 41.1818 108.554 40.9962 107.945 40.625C107.339 40.25 106.871 39.7273 106.541 39.0568C106.215 38.3826 106.053 37.5985 106.053 36.7045C106.053 35.8106 106.215 35.0227 106.541 34.3409C106.871 33.6553 107.329 33.1212 107.916 32.7386C108.507 32.3523 109.196 32.1591 109.984 32.1591C110.439 32.1591 110.888 32.2348 111.331 32.3864C111.774 32.5379 112.178 32.7841 112.541 33.125C112.905 33.4621 113.195 33.9091 113.411 34.4659C113.626 35.0227 113.734 35.7083 113.734 36.5227V37.0909H107.007V35.9318H112.371C112.371 35.4394 112.272 35 112.075 34.6136C111.882 34.2273 111.606 33.9223 111.246 33.6989C110.89 33.4754 110.469 33.3636 109.984 33.3636C109.45 33.3636 108.988 33.4962 108.598 33.7614C108.212 34.0227 107.914 34.3636 107.706 34.7841C107.498 35.2045 107.393 35.6553 107.393 36.1364V36.9091C107.393 37.5682 107.507 38.1269 107.734 38.5852C107.965 39.0398 108.286 39.3864 108.695 39.625C109.104 39.8598 109.579 39.9773 110.121 39.9773C110.473 39.9773 110.791 39.928 111.075 39.8295C111.363 39.7273 111.611 39.5758 111.82 39.375C112.028 39.1705 112.189 38.9167 112.303 38.6136L113.598 38.9773C113.462 39.4167 113.232 39.803 112.911 40.1364C112.589 40.4659 112.191 40.7235 111.717 40.9091C111.244 41.0909 110.712 41.1818 110.121 41.1818ZM116.54 32.2727L118.631 35.8409L120.722 32.2727H122.267L119.449 36.6364L122.267 41H120.722L118.631 37.6136L116.54 41H114.994L117.767 36.6364L114.994 32.2727H116.54ZM127.976 32.2727V33.4091H123.453V32.2727H127.976ZM124.771 30.1818H126.112V38.5C126.112 38.8788 126.167 39.1629 126.277 39.3523C126.391 39.5379 126.535 39.6629 126.709 39.7273C126.887 39.7879 127.074 39.8182 127.271 39.8182C127.419 39.8182 127.54 39.8106 127.635 39.7955C127.73 39.7765 127.805 39.7614 127.862 39.75L128.135 40.9545C128.044 40.9886 127.917 41.0227 127.754 41.0568C127.591 41.0947 127.385 41.1136 127.135 41.1136C126.756 41.1136 126.385 41.0322 126.021 40.8693C125.661 40.7064 125.362 40.4583 125.124 40.125C124.889 39.7917 124.771 39.3712 124.771 38.8636V30.1818Z" fill="#233D48"/>
<path d="M98.9437 57.9119V56.8182H106.58V57.9119H103.378V67H102.145V57.9119H98.9437ZM110.418 67.1591C109.682 67.1591 109.048 66.9967 108.514 66.6719C107.984 66.3438 107.574 65.8864 107.286 65.2997C107.001 64.7098 106.858 64.0237 106.858 63.2415C106.858 62.4593 107.001 61.7699 107.286 61.1733C107.574 60.5734 107.975 60.1061 108.489 59.7713C109.006 59.4332 109.609 59.2642 110.299 59.2642C110.697 59.2642 111.089 59.3305 111.477 59.4631C111.865 59.5956 112.218 59.8111 112.536 60.1094C112.854 60.4044 113.108 60.7955 113.297 61.2827C113.486 61.7699 113.58 62.3698 113.58 63.0824V63.5795H107.694V62.5653H112.387C112.387 62.1345 112.301 61.75 112.128 61.4119C111.959 61.0739 111.717 60.8071 111.403 60.6115C111.091 60.416 110.723 60.3182 110.299 60.3182C109.831 60.3182 109.427 60.4342 109.086 60.6662C108.748 60.8949 108.488 61.1932 108.305 61.5611C108.123 61.929 108.032 62.3234 108.032 62.7443V63.4205C108.032 63.9972 108.131 64.486 108.33 64.8871C108.532 65.2848 108.812 65.5881 109.17 65.7969C109.528 66.0024 109.944 66.1051 110.418 66.1051C110.726 66.1051 111.005 66.062 111.253 65.9759C111.505 65.8864 111.722 65.7538 111.905 65.5781C112.087 65.3991 112.228 65.1771 112.327 64.9119L113.461 65.2301C113.341 65.6146 113.141 65.9527 112.859 66.2443C112.577 66.5327 112.229 66.758 111.815 66.9205C111.401 67.0795 110.935 67.1591 110.418 67.1591ZM116.035 59.3636L117.864 62.4858L119.694 59.3636H121.046L118.58 63.1818L121.046 67H119.694L117.864 64.0369L116.035 67H114.683L117.109 63.1818L114.683 59.3636H116.035ZM126.041 59.3636V60.358H122.084V59.3636H126.041ZM123.237 57.5341H124.411V64.8125C124.411 65.1439 124.459 65.3925 124.555 65.5582C124.654 65.7206 124.78 65.83 124.933 65.8864C125.088 65.9394 125.253 65.9659 125.425 65.9659C125.554 65.9659 125.66 65.9593 125.743 65.946C125.826 65.9295 125.892 65.9162 125.942 65.9062L126.181 66.9602C126.101 66.9901 125.99 67.0199 125.847 67.0497C125.705 67.0829 125.524 67.0994 125.306 67.0994C124.974 67.0994 124.649 67.0282 124.331 66.8857C124.016 66.7431 123.754 66.526 123.546 66.2344C123.34 65.9427 123.237 65.5748 123.237 65.1307V57.5341Z" fill="#233D48"/>
<path d="M100.88 82.2102V81.2727H107.426V82.2102H104.681V90H103.625V82.2102H100.88ZM110.716 90.1364C110.085 90.1364 109.541 89.9972 109.083 89.7188C108.629 89.4375 108.278 89.0455 108.031 88.5426C107.787 88.0369 107.664 87.4489 107.664 86.7784C107.664 86.108 107.787 85.517 108.031 85.0057C108.278 84.4915 108.622 84.0909 109.062 83.804C109.505 83.5142 110.022 83.3693 110.613 83.3693C110.954 83.3693 111.291 83.4261 111.623 83.5398C111.956 83.6534 112.258 83.8381 112.531 84.0938C112.804 84.3466 113.021 84.6818 113.183 85.0994C113.345 85.517 113.426 86.0312 113.426 86.642V87.0682H108.38V86.1989H112.403C112.403 85.8295 112.329 85.5 112.181 85.2102C112.037 84.9205 111.829 84.6918 111.559 84.5241C111.292 84.3565 110.977 84.2727 110.613 84.2727C110.213 84.2727 109.866 84.3722 109.574 84.571C109.284 84.767 109.061 85.0227 108.904 85.3381C108.748 85.6534 108.67 85.9915 108.67 86.3523V86.9318C108.67 87.4261 108.755 87.8452 108.926 88.1889C109.099 88.5298 109.339 88.7898 109.646 88.9688C109.953 89.1449 110.309 89.233 110.716 89.233C110.98 89.233 111.218 89.196 111.431 89.1222C111.647 89.0455 111.833 88.9318 111.99 88.7812C112.146 88.6278 112.267 88.4375 112.352 88.2102L113.324 88.483C113.221 88.8125 113.049 89.1023 112.808 89.3523C112.566 89.5994 112.268 89.7926 111.913 89.9318C111.558 90.0682 111.159 90.1364 110.716 90.1364ZM115.53 83.4545L117.098 86.1307L118.666 83.4545H119.825L117.712 86.7273L119.825 90H118.666L117.098 87.4602L115.53 90H114.371L116.45 86.7273L114.371 83.4545H115.53ZM124.107 83.4545V84.3068H120.715V83.4545H124.107ZM121.703 81.8864H122.709V88.125C122.709 88.4091 122.75 88.6222 122.833 88.7642C122.918 88.9034 123.026 88.9972 123.157 89.0455C123.29 89.0909 123.431 89.1136 123.578 89.1136C123.689 89.1136 123.78 89.108 123.851 89.0966C123.922 89.0824 123.979 89.071 124.022 89.0625L124.226 89.9659C124.158 89.9915 124.063 90.017 123.941 90.0426C123.819 90.071 123.664 90.0852 123.476 90.0852C123.192 90.0852 122.914 90.0241 122.641 89.902C122.371 89.7798 122.147 89.5938 121.968 89.3438C121.792 89.0938 121.703 88.7784 121.703 88.3977V81.8864Z" fill="#233D48"/>
</svg>
