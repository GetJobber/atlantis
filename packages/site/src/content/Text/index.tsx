import Content from "@atlantis/docs/components/Text/Text.stories.mdx";
import Props from "./Text.props.json";
import MobileProps from "./Text.props-mobile.json";
import { ContentExport } from "../../types/content";

export default {
  content: () => <Content />,
  props: Props,
  mobileProps: MobileProps,
  component: {
    element: `<Text>Text</Text>`,
    mobileElement: `<Text>Text</Text>`,
  },
  title: "Text",
  links: [
    {
      label: "Storybook",
      url: "http://localhost:6006/?path=/docs/components-utilities-Text-web--docs",
    },
  ],
} as const satisfies ContentExport;
