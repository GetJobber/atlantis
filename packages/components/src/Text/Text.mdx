---
menu: Components
---

import { Playground, Props } from "docz";
import { ComponentStatus } from "@jobber/docx";
import { Text } from ".";
import { Content } from "../Content/";

# Text

<ComponentStatus stage="ready" responsive="yes" accessible="yes" />

Text is used to add an additional visual meaning on your content

```ts
import { Text } from "@jobber/components/Text";
```

<Playground>
  <Text>
    "640K is more memory than anyone will ever need on a computer" ‚Äì Not Bill
    Gates
  </Text>
</Playground>

## Props

<Props of={Text} />

---

## Variations

### Default

Use this as the body text

<Playground>
  <Text>Ask the information you need upfront from clients and new leads</Text>
</Playground>

### Subdued

De-emphasize a piece of text that is less important to the user

<Playground>
  <Text variation="subdued">Job note linked to related invoice</Text>
</Playground>

### Feedback message

<Playground>
  <Text variation="success">Invoice sent</Text>
  <br />
  <Text variation="error">Name is required</Text>
  <br />
  <Text variation="warn">Your message is over 160 characters</Text>
  <br />
  <Text variation="info">
    Drag to rearrange the order that fields show up in Jobber
  </Text>
</Playground>

## maxLines

Used to truncate large text with `...` based on line wrapping, such that the
total number of lines does not exceed the preset. This is an optional prop. If
left undefined all the contents are shown.

<Playground>
  <Content>
    <Text maxLines="single">
      This will show 1 line. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque.
    </Text>

    <Text maxLines="small">
      This will show 2 lines. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla.
    </Text>

    <Text maxLines="base">
      This will show 4 lines. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla.
    </Text>

    <Text maxLines="large">
      This will show 8 lines. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla. Etiam
      eu lacus nibh. Donec tristique lacus magna, vulputate tristique lacus
      blandit vitae. Aenean vitae commodo metus. Fusce eu risus quis orci pharetra
      consequat nec nec libero.
    </Text>

    <Text maxLines="larger">
      This will show 16 lines. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla. Etiam
      eu lacus nibh. Donec tristique lacus magna, vulputate tristique lacus
      blandit vitae. Aenean vitae commodo metus. Fusce eu risus quis orci pharetra
      consequat nec nec libero. Lorem ipsum dolor sit amet, consectetur adipiscing
      elit. Vestibulum nec pulvinar nunc. Suspendisse nec eros pretium, rutrum
      purus sit amet, finibus ante. Donec pretium condimentum scelerisque. Duis eu
      ligula nec metus suscipit feugiat.Etiam sapien sapien, mattis eu tincidunt
      quis, pretium sed metus. Maecenas quis dolor lacinia libero rhoncus
      fringilla. Cras mi ante, euismod nec tortor in, tempus mollis nulla. Etiam
      eu lacus nibh. Donec tristique lacus magna, vulputate tristique lacus
      blandit vitae. Aenean vitae commodo metus. Fusce eu risus quis orci pharetra
      consequat nec nec libero.
    </Text>

  </Content>
</Playground>

## Sizes

`Text` can be used with different sizes.

### Base

`Base` is the default size and is optimized for the widest range of uses,
including:

- Long paragraphs of text content
- Descriptions of subsequent UI elements
- Labels on UI elements such as checkboxes and radio buttons
- Freeform user-generated content

<Playground>
  <Text size="base">
    Stay organized when creating estimates, work orders, and billing documents.
  </Text>
</Playground>

### Large

Typically `large` is to be used as an introductory paragraph that gives
important information about a page, and requires greater prominence.

<Playground>
  <Text size="large">
    Attract new customers by asking happy clients for a Facebook recommendation.
  </Text>
</Playground>

### Small

The `small` size is intended to be used as helper content on things like
checkboxes, radio buttons, inputs or any other UI elements that need help
provided.

<Playground>
  <Text size="small">What is this, a font for üêús?</Text>
</Playground>
