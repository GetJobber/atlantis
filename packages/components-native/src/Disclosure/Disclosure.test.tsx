import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import { ReactTestInstance } from "react-test-renderer";
import { Disclosure } from ".";
import { Text } from "../Text";

jest.mock("react-native-svg", () => {
  return {
    __esModule: true,
    ...jest.requireActual("react-native-svg"),
    Path: "Path",
    default: "SVGMock",
  };
});

function fireLayoutEvent(disclosureContent: ReactTestInstance) {
  fireEvent(disclosureContent, "onLayout", {
    nativeEvent: {
      layout: {
        height: 100,
      },
    },
  });
}

it("renders a Collapsable item with a header and a content when open is true", () => {
  const disclosure = render(
    <Disclosure
      header={<Text>This is the header</Text>}
      content={<Text>This is the content</Text>}
      open={true}
      isEmpty={false}
      onToggle={() => {
        return;
      }}
    />,
  );
  fireLayoutEvent(disclosure.getByTestId("content"));
  expect(disclosure).toMatchSnapshot();
});

it("renders a Collapsable item with a header and with a content of size 0 when closed is false", () => {
  const disclosure = render(
    <Disclosure
      header={<Text>This is the header</Text>}
      content={<Text>This is the content</Text>}
      open={false}
      isEmpty={false}
      onToggle={() => {
        return;
      }}
    />,
  );
  fireLayoutEvent(disclosure.getByTestId("content"));
  expect(disclosure).toMatchSnapshot();
});

it("should not render the caret when the collapsable item is empty", () => {
  const disclosure = render(
    <Disclosure
      header={<Text>This is the header</Text>}
      content={<Text>This is the content</Text>}
      open={false}
      isEmpty={true}
      onToggle={() => {
        return;
      }}
    />,
  );
  expect(disclosure).toMatchSnapshot();
});
