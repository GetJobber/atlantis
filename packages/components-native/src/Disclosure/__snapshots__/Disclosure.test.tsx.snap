// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Disclosure with a header and a content when open is true 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessibilityRole="text"
        adjustsFontSizeToFit={false}
        allowFontScaling={true}
        collapsable={false}
        maxFontSizeMultiplier={3.125}
        selectable={true}
        selectionColor="hsl(86, 100%, 46%)"
        style={
          [
            {
              "fontFamily": "inter-regular",
            },
            {
              "color": "hsl(198, 35%, 21%)",
            },
            {
              "textAlign": "left",
            },
            {
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "letterSpacing": 0,
            },
          ]
        }
      >
        This is the header
      </Text>
      <View
        style={
          [
            {
              "transform": [
                {
                  "rotateZ": "0deg",
                },
              ],
            },
          ]
        }
      >
        <SVGMock
          style={
            {
              "display": "flex",
              "fill": "{color.icon}",
              "height": 24,
              "verticalAlign": "middle",
              "width": 24,
            }
          }
          testID="arrowUp"
          viewBox="0 0 24 24"
        >
          <Path
            d="M16.297 14.709a.996.996 0 0 0 1.41-.001.994.994 0 0 0 0-1.41l-5-5.005a.998.998 0 0 0-1.415 0l-5 5a.994.994 0 0 0 0 1.41.996.996 0 0 0 1.411.001L12 10.416l4.297 4.293Z"
            fill="{color.icon}"
          />
        </SVGMock>
      </View>
    </View>
  </View>
  <RCTScrollView
    collapsable={false}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    style={
      {
        "height": 100,
        "paddingTop": 8,
      }
    }
  >
    <View>
      <View
        onLayout={[Function]}
        testID="content"
      >
        <Text
          accessibilityRole="text"
          adjustsFontSizeToFit={false}
          allowFontScaling={true}
          collapsable={false}
          maxFontSizeMultiplier={3.125}
          selectable={true}
          selectionColor="hsl(86, 100%, 46%)"
          style={
            [
              {
                "fontFamily": "inter-regular",
              },
              {
                "color": "hsl(198, 35%, 21%)",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 16,
                "lineHeight": 20,
              },
              {
                "letterSpacing": 0,
              },
            ]
          }
        >
          This is the content
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders a Disclosure with a header and with a content of size 0 when closed is false 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessibilityRole="text"
        adjustsFontSizeToFit={false}
        allowFontScaling={true}
        collapsable={false}
        maxFontSizeMultiplier={3.125}
        selectable={true}
        selectionColor="hsl(86, 100%, 46%)"
        style={
          [
            {
              "fontFamily": "inter-regular",
            },
            {
              "color": "hsl(198, 35%, 21%)",
            },
            {
              "textAlign": "left",
            },
            {
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "letterSpacing": 0,
            },
          ]
        }
      >
        This is the header
      </Text>
      <View
        style={
          [
            {
              "transform": [
                {
                  "rotateZ": "-180deg",
                },
              ],
            },
          ]
        }
      >
        <SVGMock
          style={
            {
              "display": "flex",
              "fill": "{color.icon}",
              "height": 24,
              "verticalAlign": "middle",
              "width": 24,
            }
          }
          testID="arrowUp"
          viewBox="0 0 24 24"
        >
          <Path
            d="M16.297 14.709a.996.996 0 0 0 1.41-.001.994.994 0 0 0 0-1.41l-5-5.005a.998.998 0 0 0-1.415 0l-5 5a.994.994 0 0 0 0 1.41.996.996 0 0 0 1.411.001L12 10.416l4.297 4.293Z"
            fill="{color.icon}"
          />
        </SVGMock>
      </View>
    </View>
  </View>
  <RCTScrollView
    collapsable={false}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    style={
      {
        "height": 0,
        "paddingTop": 8,
      }
    }
  >
    <View>
      <View
        onLayout={[Function]}
        testID="content"
      >
        <Text
          accessibilityRole="text"
          adjustsFontSizeToFit={false}
          allowFontScaling={true}
          collapsable={false}
          maxFontSizeMultiplier={3.125}
          selectable={true}
          selectionColor="hsl(86, 100%, 46%)"
          style={
            [
              {
                "fontFamily": "inter-regular",
              },
              {
                "color": "hsl(198, 35%, 21%)",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 16,
                "lineHeight": 20,
              },
              {
                "letterSpacing": 0,
              },
            ]
          }
        >
          This is the content
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`should not render the caret when the Disclosure is empty 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessibilityRole="text"
        adjustsFontSizeToFit={false}
        allowFontScaling={true}
        collapsable={false}
        maxFontSizeMultiplier={3.125}
        selectable={true}
        selectionColor="hsl(86, 100%, 46%)"
        style={
          [
            {
              "fontFamily": "inter-regular",
            },
            {
              "color": "hsl(198, 35%, 21%)",
            },
            {
              "textAlign": "left",
            },
            {
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "letterSpacing": 0,
            },
          ]
        }
      >
        This is the header
      </Text>
    </View>
  </View>
  <RCTScrollView
    collapsable={false}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    style={
      {
        "height": 0,
        "paddingTop": 8,
      }
    }
  >
    <View>
      <View
        onLayout={[Function]}
        testID="content"
      >
        <Text
          accessibilityRole="text"
          adjustsFontSizeToFit={false}
          allowFontScaling={true}
          collapsable={false}
          maxFontSizeMultiplier={3.125}
          selectable={true}
          selectionColor="hsl(86, 100%, 46%)"
          style={
            [
              {
                "fontFamily": "inter-regular",
              },
              {
                "color": "hsl(198, 35%, 21%)",
              },
              {
                "textAlign": "left",
              },
              {
                "fontSize": 16,
                "lineHeight": 20,
              },
              {
                "letterSpacing": 0,
              },
            ]
          }
        >
          This is the content
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
